src/protocols/bdx/BdxMessages.h

```diff
@@ -312,6 +315,26 @@ template <>
 struct MessageTypeTraits<bdx::MessageType>
 {
     static constexpr const Protocols::Id & ProtocolId() { return BDX::Id; }
+
+    static auto GetTypeToNameTable()
+    {
+        static const std::array<MessageTypeNameLookup, 10> typeToNameTable = {
+            {
+                { bdx::MessageType::SendInit, "SendInit" },
+                { bdx::MessageType::SendAccept, "SendAccept" },
+                { bdx::MessageType::ReceiveInit, "ReceiveInit" },
+                { bdx::MessageType::ReceiveAccept, "ReceiveAccept" },
+                { bdx::MessageType::BlockQuery, "BlockQuery" },
+                { bdx::MessageType::Block, "Block" },
+                { bdx::MessageType::BlockEOF, "BlockEOF" },
+                { bdx::MessageType::BlockAck, "BlockAck" },
+                { bdx::MessageType::BlockAckEOF, "BlockAckEOF" },
+                { bdx::MessageType::BlockQueryWithSkip, "BlockQueryWithSkip" },
+            },
+        };
+
+        return &typeToNameTable;
+    }
```

Comment:
It's odd that logging code would be in the basic interface header of BdxMessages... Would have been good to leave inside the logging layer code

Reply:
I think it's better to keep these co-located with their enum equivalents so that there's less chance of error, and that changes to these protocols end up happening in the same file.

It also makes it more scalable when we add new protocols (though we need a better scheme for these to get picked up than than the current if/else scheme in Protocols.cpp, but that's for post 1.0)...